test_common/harness/kernelHelpers.h

```diff
@@ -121,6 +126,21 @@ cl_device_fp_config get_default_rounding_mode( cl_device_id device );
         return 0;    \
     }
 
+#define PASSIVE_REQUIRE_FP16_SUPPORT(device)                            \
+    if (!is_extension_available(device, "cl_khr_fp16"))                 \
+    {                                                                   \
+        log_info("\n\tNote: device does not support fp16. Skipping test...\n"); \
+        return 0;                                                       \
+    }
+
+#define PASSIVE_REQUIRE_IL_PROGRAM_SUPPORT( device )    \
```

Comment:
this is new to support SPIR-V being an extension where no such check is necessary in 2.1